server/gfmd/db_access.c

```diff
@@ -2188,3 +2188,114 @@ db_mdhost_load(void *closure,
 	gfarm_mutex_unlock(&db_access_mutex, diag, DB_ACCESS_MUTEX_DIAG);
 	return (e);
 }
+
+void *
+db_user_auth_dup(const struct db_user_auth_arg *ua, size_t size)
+{
+	struct db_user_auth_arg *r;
+	size_t usize = strlen(ua->username) + 1;
+	size_t msize = strlen(ua->auth_method) + 1;
+	size_t asize = strlen(ua->auth_user_id) + 1;
+	size_t sz;
+	int overflow = 0;
+
+#ifdef __GNUC__ /* workaround gcc warning: might be used uninitialized */
+	r = NULL;
+#endif
+	sz = gfarm_size_add(&overflow, size, usize + msize + asize);
+	if (!overflow)
+		r = malloc(sz);
+	if (overflow || r == NULL) {
+		gflog_debug(GFARM_MSG_1002005,
```

Comment:
change message GFARM_MSG_100205 to GFARM_MSG_UNFIXED,
@otatebe san will assign a unique message number to GFARM_MSG_UNFIXED
by using "make msgno" (makes/assign_msgno.pl) during the release process.